src/RadioButton.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { color } from 'styled-system'
+import Icon from './Icon'
+import Text from './Text'
+
+const RadioButton = ({
+  className,
+  theme,
+  onClick,
+  isSelected,
+  disabled,
+  radioButtonText
+}) => {
+  const radioIconName = isSelected ? 'radioFilled' : 'radio'
+  const radioIconColor = isSelected ? 'blue' : 'borderGray'
+
+  function handleClick() {
+    if (disabled || isSelected) return
+    onClick()
+  }
+
+  return (
+    <RadioButtonWrap
+      className={className}
+      onClick={handleClick}
+      isSelected={isSelected}
+      disabled={disabled}
+      theme={theme}
```

Comment:
Whatever the `<input>` element is should probably take the full `props` object, or rest props if some stylistic props need to be used. In this case `isSelected` looks like a proxy for the `checked` attribute, and `handleClick` is renaming `onClick` – React has built-in props for handling form elements and we should stick to those for HTML semantics, React's internal prop-type checking, familiarity with existing APIs, and for interoperability with other libraries (e.g. validation)